api/pkg/apis/v1alpha1/managers/campaigncontainers/campaign-container-manager.go

```diff
@@ -0,0 +1,181 @@
+	observ_utils "github.com/eclipse-symphony/symphony/coa/pkg/apis/v1alpha2/observability/utils"
+)
+
+var log = logger.NewLogger("coa.runtime")
+
+type CampaignContainersManager struct {
+	managers.Manager
+	StateProvider states.IStateProvider
+}
+
+func (s *CampaignContainersManager) Init(context *contexts.VendorContext, config managers.ManagerConfig, providers map[string]providers.IProvider) error {
+	err := s.Manager.Init(context, config, providers)
+	if err != nil {
+		return err
+	}
+	stateprovider, err := managers.GetStateProvider(config, providers)
+	if err == nil {
+		s.StateProvider = stateprovider
+	} else {
+		return err
+	}
+	return nil
+}
+
+func (t *CampaignContainersManager) DeleteState(ctx context.Context, name string, namespace string) error {
+	ctx, span := observability.StartSpan("CampaignContainersManager", ctx, &map[string]string{
+		"method": "DeleteState",
+	})
+	var err error = nil
+	defer observ_utils.CloseSpanWithError(span, &err)
+
+	err = t.StateProvider.Delete(ctx, states.DeleteRequest{
+		ID: name,
+		Metadata: map[string]interface{}{
+			"namespace": namespace,
+			"group":     model.SolutionGroup,
+			"version":   "v1",
+			"resource":  "campaigncontainers",
+			"kind":      "CampaignContainer",
+		},
+	})
+	return err
+}
+
+func (t *CampaignContainersManager) UpsertState(ctx context.Context, name string, state model.CampaignContainerState) error {
+	ctx, span := observability.StartSpan("CampaignContainersManager", ctx, &map[string]string{
+		"method": "UpsertState",
+	})
+	var err error = nil
+	defer observ_utils.CloseSpanWithError(span, &err)
+
+	if state.ObjectMeta.Name != "" && state.ObjectMeta.Name != name {
+		return v1alpha2.NewCOAError(nil, fmt.Sprintf("Name in metadata (%s) does not match name in request (%s)", state.ObjectMeta.Name, name), v1alpha2.BadRequest)
+	}
+	state.ObjectMeta.FixNames(name)
+
+	body := map[string]interface{}{
+		"apiVersion": model.WorkflowGroup + "/v1",
+		"kind":       "CampaignContainer",
+		"metadata":   state.ObjectMeta,
+		"spec":       state.Spec,
+	}
+
+	upsertRequest := states.UpsertRequest{
+		Value: states.StateEntry{
+			ID:   name,
+			Body: body,
+			ETag: "",
+		},
+		Metadata: map[string]interface{}{
+			"namespace": state.ObjectMeta.Namespace,
+			"group":     model.WorkflowGroup,
+			"version":   "v1",
+			"resource":  "campaigncontainers",
+			"kind":      "CampaignContainer",
+		},
+	}
+	_, err = t.StateProvider.Upsert(ctx, upsertRequest)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (t *CampaignContainersManager) ListState(ctx context.Context, namespace string) ([]model.CampaignContainerState, error) {
+	ctx, span := observability.StartSpan("CampaignContainersManager", ctx, &map[string]string{
+		"method": "ListState",
+	})
+	var err error = nil
+	defer observ_utils.CloseSpanWithError(span, &err)
+
+	listRequest := states.ListRequest{
+		Metadata: map[string]interface{}{
+			"version":   "v1",
+			"group":     model.SolutionGroup,
```

Comment:
Wrong group. You can define the group, resource, name, kind as property in manager, then we can keep them consistent.